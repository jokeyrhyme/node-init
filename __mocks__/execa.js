/* @flow */
'use strict';

let result /* : Object */ = {};

module.exports = async () /* : Promise<Object> */ => {
  return result;
};

module.exports.__setResult = (r /* : Object */) => {
  result = r;
};
