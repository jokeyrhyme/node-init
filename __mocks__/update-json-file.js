/* @flow */
'use strict';

let pkg /* : Object */ = {};

module.exports = async (
  pkgPath /* : string */,
  updater /* : (value: any) => Promise<any> */,
) /* : Promise<void> */ => {
  await updater(pkg);
};

module.exports.__setPackage = (p /* : Object */) => {
  pkg = p;
};
