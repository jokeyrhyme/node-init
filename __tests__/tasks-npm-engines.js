'use strict';

jest.mock('process');
global.process = require('process');

jest.mock('execa');
jest.mock('update-json-file');

const { fn } = require('../lib/tasks/npm-engines.js');

const TEN_SECONDS = 10 * 1e3;
jest.setTimeout(TEN_SECONDS);

test('fn() versus empty package.json, new Node.js, new npm', async () => {
  process.version = '8.12.0';
  const pkg = {};
  const result = { stdout: '6.4.1' };
  require('execa').__setResult(result);
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() versus empty package.json, old Node.js, old npm', async () => {
  process.version = '8.8.1';
  const pkg = {};
  const result = { stdout: '5.4.2' };
  require('execa').__setResult(result);
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});

test('fn() versus package.json with old engines, new Node.js, new npm', async () => {
  process.version = '10.15.0';
  const pkg = {
    engines: {
      node: '>=10.10',
      npm:  '>=4',
    },
  };
  const result = { stdout: '6.4.1' };
  require('execa').__setResult(result);
  require('update-json-file').__setPackage(pkg);

  await fn({ cwd: 'some/path' });

  expect(pkg).toMatchSnapshot();
});
