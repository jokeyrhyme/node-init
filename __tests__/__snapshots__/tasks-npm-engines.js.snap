// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fn() versus empty package.json, new Node.js, new npm 1`] = `
Object {
  "engines": Object {
    "node": ">=8",
    "npm": ">=6",
  },
}
`;

exports[`fn() versus empty package.json, old Node.js, old npm 1`] = `
Object {
  "engines": Object {
    "node": ">=8",
    "npm": ">=5",
  },
}
`;

exports[`fn() versus package.json with old engines, new Node.js, new npm 1`] = `
Object {
  "engines": Object {
    "node": ">=10.10",
    "npm": ">=6",
  },
}
`;
